alibi_detect/saving/schemas.py

```diff
@@ -668,6 +681,8 @@ class MMDDriftConfig(DriftDetectorConfig):
     n_permutations: int = 100
     batch_size_permutations: int = 1000000
     device: Optional[Literal['cpu', 'cuda']] = None
+    # Validators
+    _validate_flavour = validator('backend', allow_reuse=True, pre=False)(validate_framework)
```

Comment:
Can this be abstracted out into a base class?

Reply:
For clarification, you mean adding something like a `DriftDetectorWithBackendConfig` base pydantic model, similar to the existing `DriftDetectorConfig`, but with the `_validate_framework` validator, and then having all pydantic models such as `MMDDriftConfig` inherit from this? 

That might work, the only thing I am unsure about is that the `backend: Literal['tensorflow', 'pytorch'] = 'tensorflow'` field needs to be defined in the specific detector pydantic models, because different detectors have different `backend`'s. I'm not sure if having the `validate_framework` validator in the base model will work in that case, because `backend` isn't defined in the base model. Will look into it...

Reply:
w.r.t. clarification: yes exactly.

Reply:
Cheers for this suggestion, https://github.com/SeldonIO/alibi-detect/pull/656/commits/11c760944e80b08090a53d7b0fb53e3bb25e5aac moves `validate_framework` to `DetectorConfig`. This just required setting `check_fields=False` so that pydantic does not throw an error because `backend` is not present in `DetectorConfig`. If `backend` is present the detector config (e.g. `MMDDriftConfig`), the validator is applied. If it isn't present (e.g. `KSDriftConfig`), the validator isn't applied.

A bit cleaner I think!

Reply:
Nice!